### PR TITLE
fix: Keyboard shows up automatically when search location fragment opens.

### LINKED ISSUE #928: Keyboard does not show up automatically in search location fragment.
**Describe the bug**

When the search location fragment opens, the keyboard does not show up automatically.

**To Reproduce**
1. Goto search location fragment.
2. The keyboard does not automatically. The user has to press "where" to open the keyboard.

**Expected behavior**

The keyboard should open automatically so that user can start typing right away. This will also fix #922.

**Would you like to work on the issue?**
- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #928, #922 

Changes: Keyboard pops up automatically when the search location fragment is opened. Hence the user can start typing straight away instead of first tapping "where". Now if the back press is tapped once, the keyboard goes down and when the back press is tapped again, the user is sent back to the main activity.

GIF for the change:

ezgif com-video-to-gif 5

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt [hunk 1]
```diff
@@ -34,6 +36,9 @@ class SearchLocationFragment : Fragment() {
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         rootView = inflater.inflate(R.layout.fragment_search_location, container, false)
 
+        val imm = context?.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+        imm?.showSoftInput(view, 0)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Same method is written in utils. You can call it from there. :+1:

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @liveHarshit I tried doing it that way, but then the keyboard did not show up at all.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Okay, let me check. 
What is view here, pass it. And change `InputMethodManager.SHOW_FORCED` to `InputMethodManager.RESULT_UNCHANGED_SHOWN` in utils which contains constant value 0. Then it should work.

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @liveHarshit I replaced lines 39 and 40 in searchLocationFragment with `Utils.showSoftKeyboard(context, rootView)`. In Utils also I made the change that you mentioned above. But the keyboard still does not pop up.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: `imm?.showSoftInput(view, 0)`
What is view here?

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: That is View?, not view. It is from getView().

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: Ok then change view to nullable in utils, same as context.

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: It works. I have not made any changes in the onResume() function. It is still `Utils.showSoftKeyboard(context, rootview)`

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: But remove it! It was written for the same purpose earlier.

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: I removed it, it works fine. But now when the user presses on the back button on the top of the screen, the keyboard does not go down. It stays up in the main activity and back button on the nav bar has to pressed to close the keyboard.

Comment (yashk2000) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: @liveHarshit Should I revert back to the initial commit I made? That way the keyboard was going down automatically even when the user went to the main activity by pressing the back button in the top.

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt]: > And change `InputMethodManager.SHOW_FORCED` to `InputMethodManager.RESULT_UNCHANGED_SHOWN` in utils which contains constant value 0. Then it should work.

You forgot these change. Do it and your initial changes and these changes become same,

**Comment by yashk2000** [comment]
@liveHarshit it works now. Thanks for the help!

**Pull request merged**